lib/fathead/redis_commands/parse.py

```diff
@@ -7,41 +7,58 @@
 url = "http://redis.io"
 output = "output.txt"
 
-f = open(output, "w");
+f = open(output, "w")
 
 tree = lxml.html.parse("download/raw.dat").getroot()
-commands = tree.find_class("command")
+
+"""
+Extract all <li> elements
+First 8 elements are website's meta links, so ignore them
+"""
+elements = tree.findall(".//li")[8:]
 
 data = {}
 
-for command in commands:
+# Iterate over all desired <li> elements
+for element in elements:
+
+    # Find <a> tag within this <li> element
+    link = element.find('a')
+    # Save `href` attribute for this <a> tag
+    command_url = "{}{}".format(url, link.get('href'))
 
-    for row in command.findall('a'):
-        command_url = "%s%s" % (url, row.get('href'))
-        
-        for sibling in command.itersiblings():
-            usage = ""
-            
-            for command_args in command.findall('span'):
-                usage = "%s %s" % (row.text, command_args.text.replace(' ', '').replace('\n', ' ').strip())
+    # Find an element with class='command'
+    for command in link.find_class('command'):
+        # Save command name
+        command_text = command.text.strip()
 
-            summary = "%s." % (re.sub('\.$', '', sibling.text))
+        # Find an element with class='args'
+        for span in command.find_class('args'):
```

Comment:
Yes! You're right. There will always be one element and thus this for loop can be avoided.

Actually I focussed on mainly fixing the bug rather than re-structuring the code. I'll do that as well soon.